aten/src/ATen/native/quantized/cpu/qlinear.cpp

```diff
@@ -797,14 +797,20 @@ at::Tensor PackedLinearWeightsOnednn::apply_impl(
   TORCH_CHECK(
       dim != 0,
       "qlinear (ONEDNN): input dim should be at least 1, but got 0");
-  TORCH_CHECK(input.scalar_type() == c10::ScalarType::QUInt8,
-      "qlinear (ONEDNN): data type of input should be QUint8.");
+  TORCH_CHECK(input.scalar_type() == c10::ScalarType::QUInt8 || input.scalar_type() == c10::ScalarType::QInt8,
+      "qlinear (ONEDNN): data type of input should be QUInt8 or QInt8.");
+
+#ifdef __aarch64__
+  using qtype = int8_t;
+#else
+  using qtype = uint8_t;
+#endif
 
   auto input_contig = input.expect_contiguous();
   auto& w = *(weight_.get());
   auto K = input.size(dim - 1), M = input.numel() / K, N = w.get_dim(1);
   auto input_dims = {M, K};
-  auto input_data_type = dnnl::memory::data_type::u8;
+  auto input_data_type = std::is_signed_v<qtype> ? dnnl::memory::data_type::s8 : dnnl::memory::data_type::u8;
```

Comment:
`input` Tensor already have a dtype, what if it does not match `qdtype`?

Reply:
Resolved as part of previous comment.